tests/components/insteon/test_config_flow.py

```diff
@@ -0,0 +1,703 @@
+        entry_id="abcde12345",
+        data={**MOCK_USER_INPUT_HUB_V2, CONF_HUB_VERSION: 2},
+        options={},
+    )
+    config_entry.add_to_hass(hass)
+    assert config_entry.state == config_entries.ENTRY_STATE_NOT_LOADED
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        data={**MOCK_USER_INPUT_HUB_V2, CONF_HUB_VERSION: 2},
+        context={"source": config_entries.SOURCE_USER},
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_form_select_plm(hass: HomeAssistantType):
+    """Test we set up the PLM correctly."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await _init_form(hass, PLM)
+
+    result2, mock_setup, mock_setup_entry = await _device_form(
+        hass, result["flow_id"], mock_successful_connection, MOCK_USER_INPUT_PLM
+    )
+    assert result2["type"] == "create_entry"
+    assert result2["data"] == MOCK_USER_INPUT_PLM
+
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_select_hub_v1(hass: HomeAssistantType):
+    """Test we set up the Hub v1 correctly."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await _init_form(hass, HUB1)
+
+    result2, mock_setup, mock_setup_entry = await _device_form(
+        hass, result["flow_id"], mock_successful_connection, MOCK_USER_INPUT_HUB_V1
+    )
+    assert result2["type"] == "create_entry"
+    assert result2["data"] == {
+        **MOCK_USER_INPUT_HUB_V1,
+        CONF_HUB_VERSION: 1,
+    }
+
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_select_hub_v2(hass: HomeAssistantType):
+    """Test we set up the Hub v2 correctly."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await _init_form(hass, HUB2)
+
+    result2, mock_setup, mock_setup_entry = await _device_form(
+        hass, result["flow_id"], mock_successful_connection, MOCK_USER_INPUT_HUB_V2
+    )
+    assert result2["type"] == "create_entry"
+    assert result2["data"] == {
+        **MOCK_USER_INPUT_HUB_V2,
+        CONF_HUB_VERSION: 2,
+    }
+
+    await hass.async_block_till_done()
+    assert len(mock_setup.mock_calls) == 1
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_failed_connection_plm(hass: HomeAssistantType):
+    """Test a failed connection with the PLM."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await _init_form(hass, PLM)
+
+    result2, _, _ = await _device_form(
+        hass, result["flow_id"], mock_failed_connection, MOCK_USER_INPUT_PLM
+    )
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "cannot_connect"}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)